src/test/java/com/thealgorithms/sorts/SortUtilsTest.java

```diff
@@ -67,4 +72,23 @@ void isSortedListFalse() {
         List<Integer> array3 = List.of(5, 4, 3, 2, 1);
         assertFalse(SortUtils.isSorted(array3));
     }
+
+    @ParameterizedTest
+    @MethodSource("provideArraysForSwap")
+    public <T> void testSwap(T[] array, int i, int j, T[] expected) {
+        SortUtils.swap(array, i, j);
+        assertArrayEquals(expected, array);
+    }
+
+    @ParameterizedTest
+    @MethodSource("provideArraysForSwap")
+    public <T> void testSwapFlippedIndices(T[] array, int i, int j, T[] expected) {
+        SortUtils.swap(array, j, i);
+        assertArrayEquals(expected, array);
+    }
+
+    private static Stream<Arguments> provideArraysForSwap() {
+        return Stream.of(Arguments.of(new Integer[] {1, 2, 3, 4}, 1, 2, new Integer[] {1, 3, 2, 4}), Arguments.of(new Integer[] {1, 2, 3, 4}, 0, 3, new Integer[] {4, 2, 3, 1}), Arguments.of(new Integer[] {1, 2, 3, 4}, 2, 2, new Integer[] {1, 2, 3, 4}),
+            Arguments.of(new String[] {"a", "b", "c", "d"}, 0, 3, new String[] {"d", "b", "c", "a"}), Arguments.of(new String[] {null, "b", "c", null}, 0, 3, new String[] {null, "b", "c", null}), Arguments.of(new Integer[] {}, 0, 0, new Integer[] {}));
```

Comment:
In reality we should exclude such call.

```suggestion
            Arguments.of(new String[] {"a", "b", "c", "d"}, 0, 3, new String[] {"d", "b", "c", "a"}), Arguments.of(new String[] {null, "b", "c", null}, 0, 3, new String[] {null, "b", "c", null}));
```